pkg/resource/plugin/provider_plugin.go

```diff
@@ -954,6 +955,82 @@ func (p *provider) Invoke(tok tokens.ModuleMember, args resource.PropertyMap) (r
 	return ret, failures, nil
 }
 
+// StreamInvoke dynamically executes a built-in function in the provider, which returns a stream of
+// responses.
+func (p *provider) StreamInvoke(
+	tok tokens.ModuleMember,
+	args resource.PropertyMap,
+	onNext func(resource.PropertyMap) error) ([]CheckFailure, error) {
+
+	contract.Assert(tok != "")
+
+	label := fmt.Sprintf("%s.StreamInvoke(%s)", p.label(), tok)
+	logging.V(7).Infof("%s executing (#args=%d)", label, len(args))
+
+	// Get the RPC client and ensure it's configured.
+	client, err := p.getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	// If the provider is not fully configured, return an empty property map.
+	if !p.cfgknown {
+		onNext(resource.PropertyMap{})
+		return nil, nil
+	}
+
+	margs, err := MarshalProperties(args, MarshalOptions{
+		Label:       fmt.Sprintf("%s.args", label),
+		KeepSecrets: p.acceptSecrets,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	streamClient, err := client.StreamInvoke(
+		p.ctx.Request(), &pulumirpc.InvokeRequest{Tok: string(tok), Args: margs})
+	if err != nil {
+		rpcError := rpcerror.Convert(err)
+		logging.V(7).Infof("%s failed: %v", label, rpcError.Message())
+		return nil, rpcError
+	}
+
+	for {
+		in, err := streamClient.Recv()
+		if err == io.EOF {
+			return nil, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		// Unmarshal response.
+		ret, err := UnmarshalProperties(in.GetReturn(), MarshalOptions{
+			Label:          fmt.Sprintf("%s.returns", label),
+			RejectUnknowns: true,
+			KeepSecrets:    true,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		// Check properties that failed verification.
+		var failures []CheckFailure
```

Comment:
This is a bit weird--do we expect to only get these on the first message?

Reply:
Yeah, it's weird. We're streaming `InvokeResponse` back, but because we don't have a separate, first-class structured error channel it seems like you have to cram the error conditions into the message type—which we do both here and in `Invoke`. So you end up with this weird thing.

We could just take this out. It's not clear to me which is better.

Reply:
Eh, I think we can leave it for now. Maybe add a note with what you described above.